docs/MacOS-DEVELOP.adoc

```diff
@@ -0,0 +1,132 @@
+= rdiff-backup: {page-name}
+:page-name: MacOS Development Guide
+:pp: {plus}{plus}
+:sectnums:
+:toc: macro
+
+link:.[Home,role="button round"] // *{page-name}*
+
+'''''
+
+toc::[]
+
+Some notes for developers and contributors willing to help with rdiff-backup development on macOS, or simply compile rdiff-backup from source code.
+
+== Pre-requisites
+
+Here is the list of components required to develop or compile rdiff-backup on macOS.
+
+=== Install Python
+
+On macOS, you have two main options to install Python:
+
+- Download from the official Python website (supports universal2):  
+  https://www.python.org/downloads/macos/
+- Install via Homebrew: brew install python
+
+Verify that Python was installed correctly:
+
+[source,shell]
+----
+which python3
+python3 --version
+----
+
+=== Install Required Build Tools
+
+You may need the following tools for building native libraries:
+
+[source,shell]
+----
+xcode-select --install
+----
+
+=== Build and Install librsync from Source
+
+1. Clone or download the librsync source from GitHub:  
+   https://github.com/rdiff-backup/librsync
+
+2. Build and install it:
+
+[source,shell]
+----
+wget https://github.com/librsync/librsync/releases/download/v2.3.4/librsync-2.3.4.tar.gz
+tar zxvf librsync-2.3.4.tar.gz
+cd librsync
+cmake .
+make
+sudo make install
+----
+
+This should install `librsync` in `/usr/local/lib` and headers in `/usr/local/include`.
+
+To confirm installation:
+
+[source,shell]
+----
+ls /usr/local/lib | grep rsync
+----
+
+== Build, Test and Package rdiff-backup
+
+Now you can clone and set up the rdiff-backup source tree:
+
+[source,shell]
+----
+git clone https://github.com/rdiff-backup/rdiff-backup
+cd rdiff-backup
+----
+
+=== Create and Activate Virtual Environment
+
+[source,shell]
+----
+python3 -m venv .venv
+source .venv/bin/activate
+----
+
+Ensure you're using the virtual environment's Python:
+
+[source,shell]
+----
+which python3
+----
+
+=== Set Library Path (for dynamic linker)
+
+[source,shell]
+----
+export DYLD_LIBRARY_PATH=/usr/local/lib
+----
+
+=== Install Dependencies and Editable Package
+
+[source,shell]
+----
+pip install -e .
+----
+
+This installs rdiff-backup in editable mode along with development dependencies.
+
+=== Run Tests
+
+[source,shell]
+----
+make test
+----
+
+=== Build the Package
+
+To build a source distribution and wheel:
+
+[source,shell]
+----
+python setup.py sdist bdist_wheel
```

Comment:
Directly using setup.py should be outdated. We use `pyproject-build` (which uses setup.py under the hood).